R/slice_derivation.R

```diff
@@ -11,11 +11,8 @@
 #'   A function that performs a specific derivation is expected. A derivation
 #'   adds variables or observations to a dataset. The first argument of a
 #'   derivation must expect a dataset and the derivation must return a dataset.
-#'   The function must provide the `dataset` argument and all arguments
-#'   specified in the `params()` objects passed to the `arg` argument.
-#'
-#'   Please note that it is not possible to specify `{dplyr}`
-#'   functions like `mutate()` or `summarize()`.
+#'   All expected arguments for the derivation function must be provided through
+#'   the `params()` objects passed to the `args` arguments.
```

Comment:
It took me a while to understand that.
What about?
```suggestion
#'   All expected arguments for the derivation function must be provided through
#'   the `params()` object passed to the `args` argument or be provided in _every_ `derivation_slice()`.
```

By the way, should we improve the error message?
```
> slice_derivation(
+   select(admiral_adsl, USUBJID, ARMCD, AGE, SEX),
+   derivation = derive_var_extreme_flag,
+   args = params(new_var = AGEFL, order = exprs(AGE), by_vars = exprs(ARMCD)),
+   derivation_slice(filter = SEX == "M", params(mode = "first")),
+   derivation_slice(filter = SEX == "F", params(order = exprs(USUBJID)))
+ )
Error in (function (dataset, by_vars, order, new_var, mode, true_value = "Y",  : 
  argument "mode" is missing, with no default
```

Reply:
Can do, good suggestion. What about something like:
```
<code to get list of mandatory arguments of derivation>
for (mandatory_arg in mandatory_args){
  <check if mandatory_arg is contained in args>
  < if not, check if mandatory_arg is contained in every slice>
}
```
to be placed after the `assert_list_of(slices, "derivation_slice")` line?

If ok, is there a neat way to get the list of mandatory arguments of a function?

Reply:
Sounds good.

You can retrieve the mandatory arguments by
```
 formals(derive_extreme_records) %>% purrr::keep(rlang::is_missing) %>% names()
```

Reply:
I gave it a go - probably not perfect yet (exemplified by the failing checks!!) but what do you think about the logic?

Reply:
Looks almost good (see my comments below).